webapp/app.py

```diff
@@ -30,35 +33,51 @@
 )
 
 
+def _api_request(url_path, method="GET", params=None, json=None, headers=None):
+    """
+    Make API calls to the anbox API, passing any 401 errors to Flask to handle
+    """
+
+    response = session.request(
+        method,
+        url_join(ANBOXCLOUD_API_BASE, url_path),
+        params=params,
+        json=json,
+        headers=headers,
+    )
+
+    if response.status_code == 401:
+        flask.abort(401, response.json()["error"])
+
+    response.raise_for_status()
+
+    return response.json()
+
+
 def login_required(func):
     """
     Decorator that checks if a user is logged in, and redirects
     to login page if not.
     """
 
-    @functools.wraps(func)
     def is_user_logged_in(*args, **kwargs):
-        if "openid" not in flask.session:
-            return flask.redirect("/login?next=" + flask.request.path)
+        if "authentication_token" not in flask.session:
+            return flask.redirect("/login?next_url=" + flask.request.path)
 
+        # Validate authentication token
         return func(*args, **kwargs)
 
     return is_user_logged_in
 
 
-def request_macaroon():
-    url = f"{ANBOXCLOUD_API_BASE}/1.0/token"
-    response = requests.get(
-        url=url,
-        headers={
-            "Accept": "application/json, application/hal+json",
-            "Content-Type": "application/json",
-            "Cache-Control": "no-cache",
-        },
-        params=[("provider", "usso")],
-    )
-    response.raise_for_status()
-    return response.json()
+def get_authorization_header(root, discharge):
+    """
+    Append root and discharge to the headers
+    """
+
+    value = f"Macaroon root={root}, discharge={discharge}"
+    authorization = {"Authorization": value}
+    return authorization
```

Comment:
This could easily be:

``` python3
return {"Authorization": f"Macaroon root={root}, discharge={discharge}"}
```

And given that, this is so short as to not need to be its own function. You should just write this directly wherever you're using it.

But again, you don't actually seem to be using it anywhere:

``` python3
robin@rt480:~/Projects/anbox-cloud.io (invitation-code) $ rg get_authorization_header
webapp/app.py
73:def get_authorization_header(root, discharge):
```

Reply:
Aren't we going to use this soon? Every call moving forward will need this appended, but I guess I can remove it for now.

Reply:
Ah I think you were using this to send the header in an API call where you decided this wasn't necessary right? Do you just need to clean up this function. Thanks.

Reply:
What calls would those be?

I thought all that happens after this PR is that we pass the `authorisation_token` to the template, which hands it to the JS to do the streaming demo.

Reply:
Yes please remove it for now and add it back when you need it. Although, as I mentioned, it's so trivial that if you do need it, you still shouldn't need a function.

Reply:
oh I didn't know that we are handing it to the JS. Ok Done.